index.js

```diff
@@ -95,49 +123,39 @@ function fromDateInputString(value) {
 
 // Linkify `message` string and add it to `parent` node.
 function linkifyAndAdd(parent, message) {
-    const searches = [
-        {
-            regexp: /bug (\d+)/i,
-            createLink: match => urls.bug(match[1])
-        },
-        {
-            // Shamelessly taken from Stackoverflow:
-            // https://stackoverflow.com/questions/3809401/what-is-a-good-regular-expression-to-match-a-url
-            regexp: /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/i,
-            createLink: match => match[0]
-        }
-    ];
-
-    while (true) {
-        let matches = [];
-        for (let search of searches) {
-            const match = message.match(search.regexp);
-            if (match !== null) {
-                matches.push({
-                    search,
-                    match
-                });
-            }
-        }
+    let matches = [];
+
+    for (let search of SEARCHES) {
+        const { regexp } = search;
+        let match = null;
 
-        if (matches.length === 0) {
-            // Exit if no regular expressions matched.
-            break;
+        while (match = regexp.exec(message)) {
+            matches.push({
+                search,
+                match
+            });
         }
 
-        matches.sort((a, b) => message.indexOf(a.match[0]) > message.indexOf(b.match[0]));
+        regexp.lastIndex = 0;
+    }
+
+    matches.sort((a, b) => message.indexOf(a.match[0]) > message.indexOf(b.match[0]));
```

Comment:
Maybe pre-existing tho, this will mis-handle when there are 2 matched texts with same content.

Then, given we're now using `global` flag, we can get the matching index just after `regexp.exec` call above, by `regexp.lastIndex - match[0].length`, and store it to `matches` element, and sort by that here.